exp/api/v1beta2/rosamachinepool_types.go

```diff
@@ -134,6 +140,51 @@ type RosaMachinePoolAutoScaling struct {
 	MaxReplicas int `json:"maxReplicas,omitempty"`
 }
 
+// RosaUpdateConfig specifies update configuration
+type RosaUpdateConfig struct {
+	// RollingUpdate specifies MaxUnavailable & MaxSurge number of nodes during update.
+	//
+	// +optional
+	RollingUpdate *RollingUpdate `json:"rollingUpdate,omitempty"`
+}
+
+// RollingUpdate specifies MaxUnavailable & MaxSurge number of nodes during update.
+type RollingUpdate struct {
+	// MaxUnavailable is the maximum number of nodes that can be unavailable during the update.
+	// Value can be an absolute number (ex: 5) or a percentage of desired nodes (ex: 10%).
+	// Absolute number is calculated from percentage by rounding down.
+	//
+	// MaxUnavailable can not be 0 if MaxSurge is 0.
+	//
+	// Example: when MaxUnavailable is set to 30%, old nodes can be deleted down to 70% of
+	// desired nodes immediately when the rolling update starts. Once new nodes
+	// are ready, more old nodes be deleted, followed by provisioning new nodes,
+	// ensuring that the total number of nodes available at all times during the
+	// update is at least 70% of desired nodes.
+	//
+	// +kubebuilder:validation:Pattern="^((100|[0-9]{1,2})%|[0-9]+)$"
+	// +kubebuilder:validation:XIntOrString
+	// +optional
+	MaxUnavailable intstr.IntOrString `json:"maxUnavailable,omitempty"`
```

Comment:
should be a pointer
```suggestion
	MaxUnavailable *intstr.IntOrString `json:"maxUnavailable,omitempty"`
```

Reply:
no need for pointers, we don't want to check for nil values. empty value will be submitted anyway.

Reply:
I ma pretty sure its 0 :)  have you actually tested this?

Reply:
yes Mulham, its empty string and 0 all the time, type what determine which value (that is why originally data type was string to avoid this conversation issues. OCM uses string only) unit test is updated. Another issue I found, OCM raise error if both maxUnavailable & maxSurge not set like you cannot set the maxSurge and rely on OCM to set default for maxUnavailable (like the unit-test). I will update the PR today to avoid this.